### PR TITLE
KOGITO-779: BPMN Kogito webapp optimization

### PR DESCRIPTION
**Pull request by romartin**
Hey @domhanak @hasys @tiagobento 

This PR provides several optimizations for the kogito's BPMN editor. 

See [KOGITO-779](https://issues.redhat.com/browse/KOGITO-779)

See [updates here](https://issues.redhat.com/browse/KOGITO-779?focusedCommentId=13985697&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13985697). 

I've tested this as:
* Deploying the war in an app. server (Wildfly)
* Runnning SDM via comand line
* Running SDM via IntelliJ
* The VSCode plugin generated afer this

Assembly with:
https://github.com/kiegroup/kie-wb-common/pull/3205
https://github.com/kiegroup/kogito-tooling/pull/71

Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml [hunk 1]
```diff
@@ -39,7 +39,20 @@
...
     <module>kie-wb-common-stunner-bpmn-project-backend</module>
-    <module>kie-wb-common-stunner-bpmn-kogito-runtime</module>
   </modules>
 
+  <profiles>
+    <profile>
+      <id>kogito</id>
+      <activation>
+        <property>
+          <name>!productized</name>
```
Comment (domhanak) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml]: @mbiarnes could you please clarify, do we need to modify this in some PR job definitions?

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by tiagobento** [comment]
@romartin ^

**Comment by romartin** [comment]
Hey @tiagobento , yes I did also tests locally for kogito-tooling, see the description of this PR, I updated it with all tests I did... anyway some review should please re-test it to double check :)

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml (hunk 1 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml]: hey @domhanak - this is just for skipping that module on `productized` builds, maybe it's not a mandatory thing but it's saving some build time, so IMO it does not hurt. I don't either think this is related to any PR job.. it's just each module owner who should decide what to expose. 
This is also being done in DMN, see https://github.com/kiegroup/kie-wb-common/pull/3081 ( CC @manstis )

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml (hunk 1 above)
Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml]: Yes, makes sense, I just wanted to confirm we do not skip this on PR checks.

Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/pom.xml]: Because for our repo it makes sense to have this built too.

**Comment by domhanak** [comment]
I spotted only one problem while testing. The ID and Name were `default` value instead of what was in `.bpmn` file, but I saw it reported

**Comment by domhanak** [comment]
Also, I'd like to point out, that loading time for the editor improved for me, great job :)

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/main/resources/org/kie/workbench/common/stunner/kogito/KogitoBPMNEditor.gwt.xml [hunk 2]
```diff
@@ -53,20 +53,16 @@
...
+  <!-- Only default locale support for kogito. -->
+  <set-property name="locale" value="default"/>
 
   <!-- Configure Errai to return default values for missing i18n keys -->
   <set-property name="errai.i18n.default_per_key" value="true"/>
 
   <!-- We don't need to support IE10 or older -->
   <!-- There is no "ie11" permutation. IE11 uses the Firefox one (gecko1_8) -->
-  <set-property name="user.agent" value="gecko1_8,safari"/>
+  <set-property name="user.agent" value="safari"/>
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/main/resources/org/kie/workbench/common/stunner/kogito/KogitoBPMNEditor.gwt.xml]: No Firefox support anymore?

**Comment by romartin** [comment]
Hey @domhanak , thanks for the feedback! And you're right, we're not yet using the file name, but @inodeman is actually working on that, see [KOGITO-1354](https://issues.redhat.com/browse/KOGITO-1354)

Thanks!

**Pull request merged**